ext/net/01_net.js

```diff
@@ -31,8 +32,32 @@
     return core.shutdown(rid);
   }
 
-  function resolveDns(query, recordType, options) {
-    return core.opAsync("op_dns_resolve", { query, recordType, options });
+  async function resolveDns(query, recordType, options) {
+    let cancelRid;
+    let abortHandler;
+    if (options?.signal) {
+      options.signal.throwIfAborted();
+      cancelRid = ops.op_cancel_handle();
+      abortHandler = () => core.tryClose(cancelRid);
+      options.signal[abortSignal.add](abortHandler);
+    }
+
+    try {
+      return await core.opAsync("op_dns_resolve", {
+        cancelRid,
+        query,
+        recordType,
+        options,
+      });
+    } finally {
+      if (options?.signal) {
+        options.signal[abortSignal.remove](abortHandler);
+
+        // always throw the abort error when aborted
+        options.signal.throwIfAborted();
+        core.tryClose(cancelRid);
```

Comment:
Should the `tryClose` be before `throwIfAborted` to make sure that the RID gets closed? There was just a bug reported about apparently leaking cancellation resources: Something like this might be the reason.